genie-client/src/main/java/com/netflix/genie/client/interceptor/ResponseMappingInterceptor.java

```diff
@@ -52,8 +53,13 @@ public Response intercept(final Chain chain) throws IOException {
         if (response.isSuccessful()) {
             return response;
         } else {
-            final JsonNode responseBody = mapper.readTree(response.body().string());
-            throw new GenieClientException(response.code(), response.message() + " : " + responseBody.get("message"));
+            try {
+                final JsonNode responseBody = mapper.readTree(response.body().string());
+                throw new GenieClientException(response.code(), response.message()
+                    + " : " + responseBody.get("message"));
+            } catch (JsonMappingException jme) {
```

Comment:
Catching JsonProcessingException here will encompass JsonMappingException and JsonParseException. Or catch both explicitly.